src/olympia/blocklist/models.py

```diff
@@ -298,26 +284,34 @@ def get_changes_from_block(self, block):
                 changes[prop] = (getattr(block, prop), getattr(self, prop))
         return changes
 
-    def clean(self):
-        if self.min_version > self.max_version:
-            raise ValidationError(_('Min version can not be greater than Max version'))
-
     def get_blocks_submitted(self, load_full_objects_threshold=1_000_000_000):
-        blocks = self.block_set.all().order_by('id')
-        if blocks.count() > load_full_objects_threshold:
+        blocks_qs = self.block_set.all().order_by('id')
+        load_fakes = blocks_qs.count() > load_full_objects_threshold
```

Comment:
Follow-up: I wonder if we still need that logic?

Reply:
We could maybe switch it out for full models, but use `.only()` to load a subset of the data.  Or always load full Block instances and always have a list of `FakeBlockAddonVersion` for addon versions?  Having to handle both cases _does_ add some extra complexity.

Reply:
(Note, I didn't address this.  I still could in this PR rather than in a follow-up.)

Reply:
I was wondering if that was often reached - but I guess it is because `GUID_FULL_LOAD_LIMIT` is not too high.

Reply:
Let's keep it like that.